File: 프로그래머스/1/42748. K번째수/K번째수.kt
class Solution {
    fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
        var answer = intArrayOf()
        
        for (i in commands){
            val edited = array.copyOfRange(i[0]-1, i[1])
            if (edited.isEmpty()){
                answer = answer.plus(array[i[0]-1])
            }else{
                 edited.sort()
                answer = answer.plus(edited[(i[2]-1)])
            }
        }
        return answer
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.py
def solution(array, commands):
    answer = []
    for command in commands:
        start = command[0] - 1
        end = command[1]
        k = command[2] - 1
        slicedArr = array[start:end]
        sortedArr = sorted(slicedArr)
        answer.append(sortedArr[k])
    return answer